packages/style-engine/class-wp-style-engine.php

```diff
@@ -511,21 +614,127 @@ protected static function get_individual_property_css_declarations( $style_value
  * );.
  *
  * @return array<string>|null array(
- *     'css'          => (string) A CSS ruleset or declarations block formatted to be placed in an HTML `style` attribute or tag.
- *     'declarations' => (array) An array of property/value pairs representing parsed CSS declarations.
- *     'classnames'   => (string) Classnames separated by a space.
+ *     'css'           => (string) A CSS ruleset or declarations block formatted to be placed in an HTML `style` attribute or tag.
+ *     'declarations'  => (array) An array of property/value pairs representing parsed CSS declarations.
+ *     'classnames'    => (string) Classnames separated by a space.
  * );
  */
 function wp_style_engine_get_styles( $block_styles, $options = array() ) {
-	if ( class_exists( 'WP_Style_Engine' ) ) {
-		$style_engine = WP_Style_Engine::get_instance();
-		return $style_engine->get_styles( $block_styles, $options );
+	if ( ! class_exists( 'WP_Style_Engine' ) ) {
+		return array();
+	}
+	$defaults = array(
+		'selector'                   => null,
+		'context'                    => 'block-supports',
+		'convert_vars_to_classnames' => false,
+		'enqueue'                    => false,
+	);
+
+	$options       = wp_parse_args( $options, $defaults );
+	$parsed_styles = null;
+
+	// Block supports styles.
+	if ( 'block-supports' === $options['context'] ) {
+		$parsed_styles = WP_Style_Engine::parse_block_styles( $block_styles, $options );
+	}
+
+	// Output.
+	$styles_output = array();
+
+	if ( ! $parsed_styles ) {
+		return $styles_output;
+	}
+
+	if ( ! empty( $parsed_styles['declarations'] ) ) {
+		$styles_output['css']          = WP_Style_Engine::compile_css( $parsed_styles['declarations'], $options['selector'] );
+		$styles_output['declarations'] = $parsed_styles['declarations'];
+		if ( true === $options['enqueue'] ) {
+			WP_Style_Engine::store_css_rule( $options['context'], $options['selector'], $parsed_styles['declarations'] );
+		}
+	}
+
+	if ( ! empty( $parsed_styles['classnames'] ) ) {
+		$styles_output['classnames'] = implode( ' ', array_unique( $parsed_styles['classnames'] ) );
+	}
+
+	return array_filter( $styles_output );
+}
+
+/**
+ * Global public interface method to register styles to be enqueued and rendered.
+ *
+ * @access public
+ *
+ * @param string $store_key A valid store key.
+ * @param array  $css_rules array(
+ *     'selector'         => (string) A CSS selector.
+ *     'declarations' => (boolean) An array of CSS definitions, e.g., array( "$property" => "$value" ).
+ * );.
+ *
+ * @return WP_Style_Engine_CSS_Rules_Store|null.
+ */
+function wp_style_engine_add_to_store( $store_key, $css_rules = array() ) {
+	if ( ! class_exists( 'WP_Style_Engine' ) || ! $store_key ) {
+		return null;
+	}
+
+	// Get instance here to ensure that we register hooks to enqueue stored styles.
+	$style_engine = WP_Style_Engine::get_instance();
```

Comment:
Should the hooks be registered explicitly somewhere in WordPress, rather than when first called? Or, is it safe to leave it when this is called so that it's only registered if it's needed 🤔

This is mostly an idle thought and linked to another comment, so I don't think of it as a blocker to this PR, it's definitely something we can look at in a separate follow-up if need be.

Reply:
> Should the hooks be registered explicitly somewhere in WordPress, rather than when first called? 

Do you mean outside the style engine? As [mentioned above](https://github.com/WordPress/gutenberg/pull/42452#discussion_r934145324), I get the inkling that it might be better to give the consumer control over how styles are rendered via hooks / enqueues etc.

I think it's worth exploring in a follow up. It might mean moving a bit of logic to `script-loader.php`... not sure. 


> Or, is it safe to leave it when this is called so that it's only registered if it's needed

My thinking was that, as soon as at least one store is created, then we know to register the hook. 

Whether it's safe or not in the long term I'm not so sure, but if you have doubts then it probably won't be 😄

Reply:
> Do you mean outside the style engine?

Potentially, yes. Once the style engine is merged into core, we'll need Gutenberg to be able to switch off the core version, so whichever way we go about it, we'll need to be clear on the logic. But you raise a great point about giving consumers control over how styles are rendered!

> I think it's worth exploring in a follow up. It might mean moving a bit of logic to script-loader.php... not sure

Absolutely, I think it'll be easier to figure out the desired approach in an isolated PR. Or at the very least, figure out in a separate PR whether or not it's worth going down alternative rabbitholes 😄

> My thinking was that, as soon as at least one store is created, then we know to register the hook.

Personally, I think that's a solid strategy for the style engine — it only registers what it needs. It might be that it really is the best way to go about it, for the style engine to do the registering when it needs to. Either way it'd be good for us to document this as a principle to aid future debugging.

At this stage, I'm not too fussed about which particular direction we go in — the main goal with my comments is to ensure that we've made a clear decision about what we're attempting to do, so we can then use that to guide follow-ups. I quite like the "as soon as at least one store is created, then we know to register the hook." principle, and seeing how far that gets us 👍

Reply:
I agree, it would feel better if enqueuing etc happen outside the class. But I think we can do that in a followup PR, it doesn't have to be done all at once 👍